### PR TITLE
adding queries for get articles

### PR DESCRIPTION
**Pull request by hadeelalsaadi**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -120,6 +120,42 @@ describe("/api/articles",()=>{
...
+    })
+
+    test("GET-200 response with articles filterd by topic",()=>{
+        return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({body})=>{
+            expect(body.articles.length).toBe(12)
+        })
+    })
```
Comment (nc-jo-watson) [inline __tests__/app.test.js:150]: There is another test needed here. What if I request a topic that is valid but has no articles?

File: __tests__/app.test.js [hunk 2]
```diff
@@ -120,6 +120,42 @@ describe("/api/articles",()=>{
...
+            expect(body.msg).toBe("Bad request")
+        })
+    })
+
+    test("GET-200 response with articles filterd by topic",()=>{
+        return request(app)
+        .get("/api/articles?topic=mitch")
+        .expect(200)
+        .then(({body})=>{
+            expect(body.articles.length).toBe(12)
```
Comment (nc-mark-t) [inline __tests__/app.test.js:148]: You should check that each article has topic `mitch`

Comment (nc-mark-t) [reply __tests__/app.test.js:148]: You will also need to check for a topic that exists but has no articles such as paper

**Review by nc-jo-watson — COMMENTED** [review]
Nearly there Hadeel, you have one test missing though (that I think might change the code in your model) 👀

**Review by nc-mark-t — COMMENTED** [review]
The paper test is particularly important. See my comments. Good work though 👍

**Pull request merged**